SearchWord.java

```diff
@@ -0,0 +1,50 @@
+package com.company;
+
+import com.company.dbconnection.DatabaseLink;
+
+import java.io.BufferedReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.StringTokenizer;
+
+public class SearchWord extends Thread {
+
+    private static DatabaseLink databaseLink;
+    private static String textFilePath;
+    private static String searchKeyword;
+
+    public SearchWord(String textFilePath, String searchKeyword, DatabaseLink databaseLink) {
+
+        SearchWord.textFilePath = textFilePath;
```

Comment:
```suggestion
        this.textFilePath = textFilePath;
```
Same for the other two fields. You need to remove the static from the field declarations first

Reply:
I have made these fields not static and accessed them via this keyword.